### PR TITLE
Sx24 create plug

### PR DESCRIPTION
**Pull request by azamzamy**
#62

@maggie-moheb if you could review my tests, please.
@Youmna-Salah hello code reviewer 
@AmirMaurice hey UMLer
@daliamaarek hey documenter 
@ahmad3m3 hello scenarist
This pull request is related to a previous pull request but due to accidental merge, a new pull request was made.

### CODE REVIEW

**Comment by AmirMaurice** [comment]
@azamzamy Where are  ur class diagram & wireframe ?
In sequence diagram combine the 2 arrows of choosing plug type,name & clicking the addPlug button and the bar under the user defining the time is only one bar.

**Comment by azamzamy** [comment]
@maggie-moheb the rails test has been added, and the android test (failure assertion) has been modified.

File: SmartX/app/src/main/java/com/smartx/cookies/smartx/AddPlug.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+import models.Session;
+import retrofit.Callback;
+import retrofit.RestAdapter;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+
+
+public class AddPlug extends Activity {
+
```
Comment (daliamaarek) [inline SmartX/app/src/main/java/com/smartx/cookies/smartx/AddPlug.java:27]: add yourself as an author and add the purpose of this class :football:

File: SmartX/app/src/main/java/com/smartx/cookies/smartx/AddPlug.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    private Button next;
+
+    /**
+     *
+     * @return shared preferences of the session
+     */
+    public SharedPreferences getPrefs() {
+        return prefs;
+    }
+
```
Comment (daliamaarek) [inline SmartX/app/src/main/java/com/smartx/cookies/smartx/AddPlug.java]: add documentation for setters and getters, please

File: SmartX/app/src/main/java/com/smartx/cookies/smartx/ImageAdapter.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import android.content.Context;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.GridView;
+import android.widget.ImageView;
+import android.widget.ListAdapter;
+
+/**
+ * Created by zamzamy on 4/30/15.
```
Comment (daliamaarek) [inline SmartX/app/src/main/java/com/smartx/cookies/smartx/ImageAdapter.java]: add yourself as an author, and the purpose

File: SmartX/app/src/main/java/com/smartx/cookies/smartx/ImageAdapter.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    /**
+     *
+     * @param position the position in which the item found there should have its ID returned.
+     * @returns the id of the item found at the given position.
+     */
+    public long getItemId(int position) {
+        return 0;
+    }
+
+    // create a new ImageView for each item referenced by the Adapter
```
Comment (daliamaarek) [inline SmartX/app/src/main/java/com/smartx/cookies/smartx/ImageAdapter.java]: state the parameters clearly :cookie:

File: SmartX/app/src/main/java/com/smartx/cookies/smartx/PlugIcon.java [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+import models.Plug;
+import retrofit.Callback;
+import retrofit.RestAdapter;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+
+
+public class PlugIcon extends Activity {
+
```
Comment (daliamaarek) [inline SmartX/app/src/main/java/com/smartx/cookies/smartx/PlugIcon.java:30]: author and purpose :D

File: SmartX/app/src/main/java/models/Plug.java [hunk 6]
```diff
@@ -0,0 +1,147 @@
+package models;
+/**
+ * Created by zamzamy on 4/30/15.
```
Comment (daliamaarek) [inline SmartX/app/src/main/java/models/Plug.java:3]: author and purpose here too

**Comment by daliamaarek** [comment]
add yourself as an author in myAPI too

**Comment by azamzamy** [comment]
@daliamaarek done
you have absolutely no idea how much i want to _431bf7f53c87176264062331bd3d697f

**Comment by azamzamy** [comment]
@Youmna-Salah with all due respect, I'm not changing anything. To me, this code is much more comprehensive this way.
http://programmers.stackexchange.com/questions/17305/how-do-you-use-blank-lines-in-your-code

**Comment by azamzamy** [comment]
@Youmna-Salah done

**Comment by azamzamy** [comment]
@AmirMaurice Added class diagram, modified the sequence diagram, please review

**Comment by maggie-moheb** [comment]
tests of android and rails are great :+1:

**Comment by AmirMaurice** [comment]
@azamzamy u need wireframes, in class diagrams we write the name of attribute followed by the type & for the sequence diagram you didn't combine the 2 arrows in one ( choose plug type,name & add plug button)

**Comment by azamzamy** [comment]
@AmirMaurice done

**Comment by AmirMaurice** [comment]
@azamzamy perfect :+1:

**Comment by ahmad3m3** [comment]
@azamzamy  good job :+1:

**Comment by azamzamy** [comment]
@ahmad-diaa Green button :)

**Pull request merged**